realm/realm-library/src/objectServer/java/io/realm/permissions/PermissionOffer.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.permissions;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Date;
+import java.util.UUID;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.realm.PermissionManager;
+import io.realm.annotations.Index;
+import io.realm.annotations.PrimaryKey;
+import io.realm.annotations.RealmClass;
+import io.realm.annotations.Required;
+import io.realm.internal.Util;
+import io.realm.internal.permissions.BasePermissionApi;
+
+
+/**
+ * This class represents a permission offer for a Realm that can be given to other users.
+ * When an offer is successfully created, it will be represented by an {@code offerToken} that can be sent
+ * to other users. Once they accept this token, the permissions covered by this offer will take effect for that
+ * user.
+ * <p>
+ * Permission offers can only be created by users that can manage the Realm, the offer is about.
+ *
+ * @see PermissionManager#makeOffer(PermissionOffer, PermissionManager.Callback)
+ * @see PermissionManager#acceptOffer(String, PermissionManager.Callback)
+ * @see <a href="https://realm.io/docs/realm-object-server/#permissions">Permissions description</a> for general
+ * documentation.
+ */
+
+@RealmClass
+public class PermissionOffer implements BasePermissionApi {
+
+    // Base fields
+    @PrimaryKey
+    @Required
+    private String id = UUID.randomUUID().toString();
```

Comment:
I don't think so, but a UUID has a well-defined standard format + we are already using this with the management realm.